### ISSUE TITLE
BUG: Conform to cuda toolkit/cuda driver compatibility

### ISSUE THREAD

**Issue by ilemhadri** [google/jax#11967]
### Description

TLDR: the latest `jax`/`jaxlib`  do not support GPU with `cuda-11.6.2` and nvidia driver version 470 even though this is within the [compatibility specs](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html) of the cuda toolkit. Downgrading to `cuda-11.4.2` works.I **think** upgrading the driver version would also work, but this would be a major task for a grid of our scale.

The error message is the [typical](https://github.com/google/jax/issues/5723):
```
jax.devices()
>>> [GPUDevice(0)
jax.numpy.ones(3)
>>>RuntimeError: Unknown: no kernel image is available for execution on the device
```

Setting the `xla_gpu_force_compilation_parallelism=1` works but has a performance hit.

I suspect that upgrading the driver to 500+ would fix this problem. However, according to Table 2 in the [CUDA toolkit specification](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html), cuda 470 is compatible with cuda-11.6.2.

is there a way that jax (or rather xla) could conform to the minimum requirements there, so that if NVIDIA claims that driver v.470 works with cuda v.11.6.2 then it should work for jax as well?

(Happy to address this question to XLA team directly if that would be more relevant.)

### What jax/jaxlib version are you using?

jax 0.3.16, jaxlib 0.3.15

### Which accelerator(s) are you using?

V100

### Additional System Info

Python 3.8

**Comment by hawkinsp** [comment]
Your driver is too old:

https://github.com/google/jax#pip-installation-gpu-cuda

"
You must use an NVidia driver version that is at least as new as your [CUDA toolkit's corresponding driver version](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-major-component-versions__table-cuda-toolkit-driver-versions). For example, if you have CUDA 11.4 update 4 installed, you must use NVidia driver 470.82.01 or newer if on Linux. This is a strict requirement that exists because JAX relies on JIT-compiling code; older drivers may lead to failures.
"

11.6.2 CUDA requires driver 510.47.03 or newer on Linux.

(We can probably do better than this in the future, but that's the current compatibility requirement from JAX.)

**Comment by hawkinsp** [comment]
I should add there's a workaround: take the `ptxas` from an *older* CUDA toolkit that is compatible with your driver and put it in your `PATH`. That will work around the problem.

**Comment by hawkinsp** [comment]
@ilemhadri It's actually not related to the JAX version. It's related to the version of `ptxas` in your `PATH`! If you have a new `ptxas` in your `PATH`, you need a matching driver. But as I mentioned, you can just downgrade that locally.

**Comment by ilemhadri** [comment]
Thanks @hawkinsp -- Closing this issue for now as that addresses my concerns -- with a bit of extra work, hacking the `ptxas` is feasible in our case.
